src/android/ImagePicker.java

```diff
@@ -69,62 +74,117 @@ public boolean execute(String action, final JSONArray args, final CallbackContex
     }
 
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (resultCode == Activity.RESULT_OK && data != null) {
-            ArrayList<String> fileURIs = new ArrayList<String>();
-            if (requestCode == SELECT_PICTURE) {
-                if (data.getData() != null) {
-                    Uri uri = data.getData();
-                    String path = this.copyFileToInternalStorage(uri, "");
-                    if (path.equals("-1")) {
-                        callbackContext.error(CROSS_USER_PROFILE_ACCESS_DENIED);
-                        return;
-                    }
-                    fileURIs.add(path);
-                } else {
-                    ClipData clip = data.getClipData();
-                    for (int i=0; i<clip.getItemCount(); i++) {
-                        Uri uri = clip.getItemAt(i).getUri();
-                        String path = this.copyFileToInternalStorage(uri, "");
+        new RetrieveData(requestCode, resultCode, data).execute();
+    }
+
+    class RetrieveData extends AsyncTask<Void, Void, Void> {
```

Comment:
https://developer.android.com/reference/android/os/AsyncTask
AsyncTask is deprecated as from API 30 😞 
The android doc proposes to use the java.util.concurrent Executors instead to do the background job.
Example below:
https://stackoverflow.com/a/64969640

Reply:
C'est fait